### DISCUSSION TITLE
https://klp1.com/firmware/

### DISCUSSION THREAD

**Discussion by giscus** [blakadder/klp1#4] [Announcements]
# https://klp1.com/firmware/

Kingroon KLP1 firmware image, configuration files and slicer profiles

https://klp1.com/firmware/

**Comment by xbiff** [comment]
profile not at cura 5.4.0 .

**Comment by quimnut** [comment]
orca profile here, doesn't seem to have filament change gcode, assuming that needs to be paired with some macros that aren't in the default fw
https://onedrive.live.com/?authkey=%21AJfUeKOzJS73xyI&id=D97D20DCB9569C61%212538&cid=D97D20DCB9569C61

**Comment by zentaur1ger** [comment]
Is there somewhere the firmware for the Kingroon Cheetah V2.2 available?

**Comment by quimnut** [comment]
For KLP1, the machine barcode corresponding to 2.1 does not have an X at the end, and the machine barcode corresponding to 2.2 has an X at the end.

Links for firmware https://kingroon.com/blogs/3d-print-101/how-to-falsh-the-emmc-chip-on-kingroon-kp3s-pro-v2-klp1-klipper-3d-printers

**Reply by zentaur1ger** [reply]
I know this link. But at the lower end of the site it's mentioned, that these firmware's are only usable for cheetah V1.0 and V2.0, not for V2.2

**Comment by andyvideeo** [comment]
Anyone here have the original KLP1 230 printer.cfg? The system images available all have extruder settings for the 210 version. I have calibrated mine but the original file doesnt seem to be anywhere online.

**Reply by Jpe230** [reply]
https://w.mail.qq.com/ftn/download?func=3&key=763037361b4122f2a63afa1b466154541f050e030000505e561d560155524e520507521b005150541d560f545605535e01005603076173667b7c67074c5350568e85f8c74f1b0a16300e&code=0076aacf

Taken from discord

**Reply by andyvideeo** [reply]
Thank you very much!

**Reply by andyvideeo** [reply]
How do I download though?

**Reply by andyvideeo** [reply]
Google translate says :"The file has expired or been deleted" :-(

**Comment by 3dr14n** [comment]
Did anyone here have upgraded their KLP1 230 via console? I've accessed my console and saw lots of available upgrades but I was very adamant of installing any of them since Kingroon has not documented any update or even backup procedure when everything goes wrong.

**Reply by andyvideeo** [reply]
You can upddate -BUT- you will need to flash the mainboard and the MKS-THR module.
The Mainboard part is easy, using KIAUH to make the file and pop it on an SD card. The THR (in the print head) requires some soldering on the 230.
If your machine is running correctly and you don't really know what youre doing I suggest leave it be!

**Reply by 3dr14n** [reply]
It's not really an upgrade of the whole OS. Just an update of existing packages in its OS. I did update after some modifications with the package sources and it updated fine without any issue.

**Reply by snelweg** [reply]
@3dr14n would be nice to know how you did that?

**Reply by andyvideeo** [reply]
The THR module can be flashed by adding a usb cable to the ports marked 5v, Gnd, D+ and D-. You press the boot button whilst insering the usb into the PC containing the flash file. Drag and drop.
https://photos.app.goo.gl/gB2Wpebuxd7Jm2iZ6

**Reply by karabas2011** [reply]
Can katapult firmware be applyable to  THR v2.0?
I want to avoid any wireing for future updates

**Comment by Flozilla2001** [comment]
Does anybody have a system image of the KLP1 230? I tried to update crowsnest and now it´s not possible to rollback to the version they used...
I would simply reflash the EMMC.

**Reply by snelweg** [reply]
this one works for me: https://filebin.net/t7cappahpzkmkwf4

**Comment by shawneye** [comment]
I upgraded my KLP1 230 to official klipper and the OS to Bookworm. You may need an EMMC with more space. I upgraded to a 32GB for less than $10. Make a backup of your config files and use KIAUH to uninstall everything installed. Then upgrade to bookworm (Google Armbian upgrade Buster to Bookworm). Then use KIAUH to reinstall and you will have official Klipper and upgrade any time needed. Like Anyvideo said you will need to solder some pins on the THR board but it was not difficult. See the end of this to see what to solder and for config requirements to flash both boards. I tried to update with the instructions but could not get it to work so I just upgraded to way listed above. This is for the KP3S pro but they use the same board.
https://github.com/Lebensgefahr/kingroon_kp3s_pro_v2

**Reply by karabas2011** [reply]
I upgraded my klp1 210 to bookworm (board cheetah v2.2) . I had hard time to struggle with touchscreen and choosing correct dtb and kernel versions
I did not solder anything to THR  but made custom cable as recommended. 
Nevertheless on several printers sometimes board does not connect to THR as it does on older armbian versions

**Reply by shawneye** [reply]
My touchscreen and the kingroon logo both stopped working. I'm wondering if the logo also lights the screen. I have a 5 inch screen attached to mine so I haven't looked into the tiny screen problem yet. The touch part works I just can't see it.

**Reply by shawneye** [reply]
I did the same thing. I had to solder pins to the THR(v2) board and then my cable can attach as needed.

**Reply by karabas2011** [reply]
> My touchscreen and the kingroon logo both stopped working. I'm wondering if the logo also lights the screen. I have a 5 inch screen attached to mine so I haven't looked into the tiny screen problem yet. The touch part works I just can't see it.

My display is stock MKS PI-TS35 no problem. If I connect HDMI display(for PC) I see another Kingroon logo. But 7" hdmi display shows nothing

**Reply by karabas2011** [reply]
> I did the same thing. I had to solder pins to the THR(v2) board and then my cable can attach as needed.

I wonder if katapult firmware will work
